lib/simctl.js

```diff
@@ -33,45 +54,145 @@ async function simCommand (command:string, timeout:number, args:Array = [], env
+ *                 returns non-zero return code.
+ */
+async function installApp (udid, appPath) {
   await simExec('install', 0, [udid, appPath]);
 }
 
-async function bootDevice (udid:string):void {
+/**
+ * Boot the particular Simulator if it is not running.
+ * 
+ * @param {string} udid - The UDID of an existing Simulator.
+ * @throws {Error} If the corresponding simctl subcommand command
+ *                 returns non-zero return code.
+ */
+async function bootDevice (udid) {
   await simExec('boot', 0, [udid]);
 }
 
-async function removeApp (udid:string, bundleId:string):void {
+/**
+ * Remove the particular application package from Simulator.
+ * It is required that Simulator is in _booted_ state and
+ * the application with given bundle identifier is already installed.
+ * 
+ * @param {string} udid - The UDID of an existing Simulator.
+ * @param {string} bundleId - Bundle identifier of the application,
+ *                            which is going to be removed.
+ * @throws {Error} If the corresponding simctl subcommand command
+ *                 returns non-zero return code.
+ */
+async function removeApp (udid, bundleId) {
   await simExec('uninstall', 0, [udid, bundleId]);
 }
 
-async function launch (udid:string, bundleId:string, tries:int = 5):void {
+/**
+ * Execute the particular application package on Simulator.
+ * It is required that Simulator is in _booted_ state and
+ * the application with given bundle identifier is already installed.
+ * 
+ * @param {string} udid - The UDID of an existing Simulator.
+ * @param {string} bundleId - Bundle identifier of the application,
+ *                            which is going to be removed.
+ * @param {number} tries [5] - The maximum number of retries before
+ *                             throwing an exception.
+ * @throws {Error} If the corresponding simctl subcommand command
+ *                 returns non-zero return code.
+ */
+async function launch (udid, bundleId, tries = 5) {
   await retryInterval(tries, 1000, async () => {
     await simExec('launch', 0, [udid, bundleId]);
   });
 }
 
-async function spawn (udid:string, executablePath:string, env = {}):void {
+/**
+ * Spawn the particular process on Simulator.
+ * It is required that Simulator is in _booted_ state.
+ * @name spawn
+ * 
+ * @param {string} udid - The UDID of an existing Simulator.
+ * @param {string} executablePath - The path to the process on
+ *                                  internal Simulator file system.
+ * @param {object} env [{}] - Additional environment variables mapping.
+ * @return {ExecResult} Command execution result.
+ * @throws {Error} If the corresponding simctl subcommand command
+ *                 returns non-zero return code.
+ */
+async function spawn (udid, executablePath, env = {}) {
   return await simExec('spawn', 0, [udid, executablePath], env);
 }
 
-async function spawnSubProcess (udid:string, executablePath:string, env = {}):void {
+/**
+ * @alias spawn
```

Comment:
This is not correct. `spawn` uses `exec` (which runs and returns a value) and `spawnSubProcess` creates a `SubProcess` (which does not run).

Reply:
Thanks, I'll fix that. Didn't know it myself %)